tests/integration/test_compute_flow.py

```diff
@@ -20,49 +20,115 @@
 from ocean_lib.web3_internal.currency import to_wei
 from ocean_lib.web3_internal.wallet import Wallet
 from tests.resources.ddo_helpers import (
-    get_registered_algorithm_ddo,
-    get_registered_algorithm_ddo_different_provider,
+    get_raw_algorithm,
+    get_registered_algorithm_with_access_service,
     get_registered_asset_with_access_service,
     get_registered_asset_with_compute_service,
 )
-from web3.logs import DISCARD
 
 
 @pytest.fixture
 def dataset_with_compute_service(publisher_wallet, publisher_ocean_instance):
+    """Returns a dataset with compute service.
+    Fixture is registered on chain once and can be used multiple times.
+    Reduces setup time."""
     # Dataset with compute service
     asset = get_registered_asset_with_compute_service(
         publisher_ocean_instance, publisher_wallet
     )
-
-    # verify the ddo is available in Aquarius
+    # verify the asset is available in Aquarius
     publisher_ocean_instance.assets.resolve(asset.did)
+    return asset
 
+
+@pytest.fixture
+def dataset_with_compute_service_generator(publisher_wallet, publisher_ocean_instance):
+    """Returns a new dataset each time fixture is used.
+    Useful for tests that need to update the dataset"""
+    # Dataset with compute service
+    asset = get_registered_asset_with_compute_service(
+        publisher_ocean_instance, publisher_wallet
+    )
+    # verify the asset is available in Aquarius
+    publisher_ocean_instance.assets.resolve(asset.did)
     yield asset
 
 
 @pytest.fixture
-def algorithm(publisher_wallet, publisher_ocean_instance):
+def dataset_with_compute_service_allow_raw_algo(
+    publisher_wallet, publisher_ocean_instance
+):
+    # Dataset with compute service
+    asset = get_registered_asset_with_compute_service(
+        publisher_ocean_instance, publisher_wallet, allow_raw_algorithms=True
+    )
+    # verify the asset is available in Aquarius
+    publisher_ocean_instance.assets.resolve(asset.did)
+    return asset
+
+
+@pytest.fixture
+def dataset_with_compute_service_and_trusted_algorithm(
+    publisher_wallet, publisher_ocean_instance, algorithm
+):
     # Setup algorithm meta to run raw algorithm
-    asset = get_registered_algorithm_ddo(publisher_ocean_instance, publisher_wallet)
+    asset = get_registered_asset_with_compute_service(
+        publisher_ocean_instance, publisher_wallet, trusted_algorithms=[algorithm]
+    )
     # verify the ddo is available in Aquarius
     _ = publisher_ocean_instance.assets.resolve(asset.did)
```

Comment:
please remove all _ assignments. Just call the function if you don't need its result.

Reply:
Done.